### PR TITLE
288A merge master into sqw feature branch 110421

### PR DESCRIPTION
**Pull request by cmarooney-stfc**
The sqw feature branch 288_sqw_rewrite_feature is now ready to be merged into master. As a preliminary, this PR merges master into the feature branch - the result will subsequently be merged into master.

### CODE REVIEW

**Review by abuts — APPROVED** [review]
It looks like number of tickets merged together. All changes make sense in general. Not sure can judge the details

File: horace_core/sqw/@SQWDnDBase/sqw_eval.m [hunk 1]
```diff
@@ -1,76 +1,154 @@
...
+        % If filebacked, always return file paths not objects. This stops us from
+        % leaking file-backed objects
+        if numel(opts.outfile) > 1
+            wout = opts.outfile;
+        else
+            wout = opts.outfile{1};
+        end
+    end
+    
+    end % of function sqw_eval
```
Comment (oerc0122) [inline horace_core/sqw/@SQWDnDBase/sqw_eval.m:114]: Improperly indented from here on

File: horace_core/sqw/@SQWDnDBase/sqw_eval.m [hunk 2]
```diff
@@ -1,76 +1,154 @@
...
+    
+        if outfiles_empty && opts.filebacked
+            opts.outfile = gen_unique_file_paths( ...
+                numel(win), 'horace_sqw_eval', tmp_dir(), 'sqw' ...
+            );
+        end
+    end
+    
+    %=================================================================
+
```
Comment (oerc0122) [inline horace_core/sqw/@SQWDnDBase/sqw_eval.m:154]: No terminal newline

File: horace_core/sqw/@SQWDnDBase/sqw_eval_nopix_.m [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
+    %   obj        Dataset (or array of datasets) that provides the axes and points
+    %              for the calculation
+    %
+    %   sqwfunc     Handle to function that calculates S(Q,w)
+    %   all_bins    Boolean flag wither to apply function to all bins or only those contaiing data
+    %   pars       Arguments needed by the function.
+    %
+    %=================================================================
+    
+       qw = calculate_qw_bins(obj);
```
Comment (oerc0122) [inline horace_core/sqw/@SQWDnDBase/sqw_eval_nopix_.m:18]: Improperly indented

File: horace_core/sqw/@SQWDnDBase/sqw_eval_nopix_.m [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
-   else
-       obj.data_.s = reshape(sqwfunc(qw{:}, pars{:}), size(obj.data_.s));
-   end
-   obj.data_.e = zeros(size(obj.data_.e));
-end
-
+       obj.data_.e = zeros(size(obj.data_.e));
+    end
+    
+
```
Comment (oerc0122) [inline horace_core/sqw/@SQWDnDBase/sqw_eval_nopix_.m:31]: No terminal newline

**Review by oerc0122 — COMMENTED** [review]
This seems to be a munging of several MRs, including Harry's install instructions and my error-format ADR. 

I would recommend doing a manual merge onto a local master first to ensure no duplicated files, commits or weirdness is going on here.

**Pull request merged**